package/MDAnalysis/analysis/hydrogenbonds/hbond_analysis.py

```diff
@@ -401,17 +401,25 @@ def guess_hydrogens(self,
         ag = self.u.select_atoms(select)
         hydrogens_ag = ag[
             np.logical_and.reduce((
-                ag.masses < max_mass,
-                ag.charges > min_charge,
-                ag.masses > min_mass,
+                ag.masses <= max_mass,
```

Comment:
Confusing doesn't mean it's a bug, if the behaviour changes such that the numbers will change this is a pretty big break on a science perspective. Knowing the types of places this code gets used, I'm going to take the stance that this needs at least one version of user warning?

Reply:
Tagging @MDAnalysis/coredevs for discussion.

Reply:
The docs state (my emphasis)

> This function makes use of atomic masses and atomic charges to identify which atoms are hydrogen atoms that are capable of participating in hydrogen bonding. If an atom has a mass **less than** :attr:`max_mass` and an  atomic charge **greater than** :attr:`min_charge` then it is considered capable of participating in hydrogen bonds.

so the current code does not contradict the docs, does it? Therefore, what's the justification of changing the code, especially as @IAlibay notes, people may rely on the defaults.

Maybe just making the docs clearer (the description of the kwargs) would be sufficient? Given that these are floating point comparisons, the equality part really should not matter too much.

Reply:
I now see that distinction in the notes of the methods `guess_hydrogens`, but the definition under Parameters uses inclusive language:
>         max_mass: float (optional)
>             Maximum allowed mass of a hydrogen atom.
>         min_charge: float (optional)
>             Minimum allowed charge of a hydrogen atom.

If you would prefer, I can update the language under Parameters to be consistent with the Notes section for this PR. There are a few points I would make first (With vote to change inequalities or not):
- (Nay) The papers that use this analysis normally say "within", which assumes exclusive use.
    - This only applied to distance cutoffs, not charge cutoffs.
- (Yay) *However*, the definition for the angle restriction in MDAnalysis is actually the supplementary angle of what is reported in the literature. If the literature cutoff is expected to be interpreted exclusively, then the MDAnalysis angle cutoff should be inclusive.
- (Nay) In other cases in MD cutoffs are exclusive. (pair cutoffs and coordination cutoffs)
- (Yay) The distance based selection criteria available through MDAnalysis use inclusive maximum cutoffs, but with exclusive minimum, cutoffs. 
    - To stay consistent, this would mean changing only the maximum constraints

Reply:
For 2.x just changing the docs should do it.

For 3.x we can consider changing the treatment of the boundary if people are interested in it. If so, we would
1. raise a new issue for this specific problem
2. add deprecation notes and warnings to the code
3. make the change when we release 3.0